src/differentials.jl

```diff
@@ -191,6 +235,16 @@ end
     return element, (externed, new_state)
 end
 
+unthunk(x) = x
+unthunk(x::AbstractThunk) = unthunk(x())
+
+wirtinger_primal(::Union{AbstractThunk}) =
+    throw(ArgumentError("`wirtinger_primal` is not defined for `AbstractThunk`. Call `unthunk` first."))
+wirtinger_conjugate(::Union{AbstractThunk}) =
+    throw(ArgumentError("`wirtinger_conjugate` is not defined for `AbstractThunk`. Call `unthunk` first."))
```

Comment:
Weird `Union`

Also not sure about this function but will wait and see

Reply:
Yes, the `Union` is just a relic. Probably don't need this anymore, since `Thunk`s should now never wrap `AbstractWirtinger`.